### PR TITLE
Setup project skeleton

### PR DESCRIPTION
**Pull request by pvtuan10**
This PR setups skeleton for this project, which includes:
- Maven multi-module project setup.
- Docker compose + Dockerfiles setup.
- Update README.

### CODE REVIEW

File: jenkins/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM ubuntu
```
Comment (oleg-nenashev) [inline jenkins/Dockerfile]: I would suggest using a standard `jenkins/jenkins` image and installing the plugin on the top of it.

https://github.com/oleg-nenashev/demo-jenkins-config-as-code/blob/master/Dockerfile does plugin installation using `plugins.txt`, but you can just copy the HPI to a plugin directory

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,50 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+    <parent>
+        <groupId>org.jenkins-ci.plugins</groupId>
+        <artifactId>plugin</artifactId>
```
Comment (oleg-nenashev) [inline pom.xml]: With such implementation Kafka Client plugin will be also using the Plugin POM. This Parent POM does not support JAR packaging, it's better to use https://github.com/jenkinsci/pom instead

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    </parent>
+
+    <artifactId>remoting-kafka-plugin</artifactId>
+    <name>Remoting Kafka Plugin</name>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>pom</packaging>
+
+    <properties>
+        <jenkins.version>2.7.3</jenkins.version>
+        <java.level>8</java.level>
```
Comment (oleg-nenashev) [inline pom.xml]: If you use Java 8, please use Jenkins core 2.60.x or above +

File: docker-compose.yml [hunk 4]
```diff
@@ -0,0 +1,35 @@
+version: '2'
+services:
+    remoting-kafka:
```
Comment (oleg-nenashev) [inline docker-compose.yml]: I would call it `jenkins-kafka-agent` or so

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,10 @@
+FROM ubuntu
```
Comment (oleg-nenashev) [inline Dockerfile]: Maybe it's better to move it to the `client` directory

File: pom.xml [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+
+    <licenses>
+        <license>
+            <name>MIT License</name>
+            <url>https://opensource.org/licenses/MIT</url>
+        </license>
+    </licenses>
+
+    <modules>
+        <module>client</module>
```
Comment (oleg-nenashev) [inline pom.xml]: WDYT about naming it an agent?

File: client/pom.xml [hunk 7]
```diff
@@ -0,0 +1,17 @@
...
+        <artifactId>remoting-kafka-plugin</artifactId>
+        <version>1.0-SNAPSHOT</version>
+        <relativePath>../pom.xml</relativePath>
+    </parent>
+
+    <artifactId>remoting-kafka-client</artifactId>
+    <version>1.0-SNAPSHOT</version>
+    <packaging>jar</packaging>
+    <name>Remoting Kafka Plugin Client</name>
+</project>
```
Comment (dwnusbaum) [inline client/pom.xml]: Not very important, but it's nice to have a newline at the end of all files.

File: jenkins/Dockerfile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+
+EXPOSE 8080
+
+RUN apt-get update && apt-get install -y software-properties-common
+RUN echo oracle-java8-installer shared/accepted-oracle-license-v1-1 select true | debconf-set-selections && \
```
Comment (oleg-nenashev) [inline jenkins/Dockerfile]: It's not something feasible for a public image. Better to use OpenJDK

**Comment by pvtuan10** [comment]
@oleg-nenashev I changed the structure as you suggested, not sure what happened with the CI build.

**Comment by pvtuan10** [comment]
@oleg-nenashev can we get this merge soon?

**Comment by casz** [comment]
@oleg-nenashev what prevents plugin-pom from jar packing? They both include maven-jar-plugin?

https://github.com/jenkinsci/plugin-pom/blob/8224b204a5646d4292da7b3e38876ca6fbde9c33/pom.xml#L307-L310
https://github.com/jenkinsci/pom/blob/5e24fccbaaa39ed5f4d16a6457d1820f4ca58eb5/pom.xml#L219-L222

**Pull request merged**